### PR TITLE
[Fix] bad data consistency on Customized Strategy

### PR DESCRIPTION
**Pull request by rhdong**
- Add strategy check in `insert_and_evict`

### CODE REVIEW

File: include/merlin/core_kernels/upsert_and_evict.cuh [hunk 1]
```diff
@@ -70,6 +70,8 @@ __global__ void upsert_and_evict_kernel_with_io_core(
     } while (occupy_result == OccupyResult::CONTINUE);
 
     if (occupy_result == OccupyResult::REFUSED) {
+      evicted_keys[key_idx] = insert_key;
+      evicted_scores[key_idx] = insert_score;
```
Comment (jiashuy) [inline include/merlin/core_kernels/upsert_and_evict.cuh:74]: So If a KV-pair with a small score than all KV-pairs in the bucket, the new KV-pair will be also ignored, right?

**Review by jiashuy — APPROVED** [review]

File: include/merlin/core_kernels/upsert_and_evict.cuh (hunk 1 above)
Comment (rhdong) [reply include/merlin/core_kernels/upsert_and_evict.cuh:74]: Yeah, will be directly put into the evicted buffer. Let `data` itself make a decision.

**Pull request merged**